models/Formatter/ItemResponseVariableSplitter.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoResultServer\models\Formatter;
+
+use oat\oatbox\service\ConfigurableService;
+
+class ItemResponseVariableSplitter extends ConfigurableService
+{
+    public function splitByAttempt(array $itemVariables): array
+    {
+        $attempts = [];
+        foreach ($itemVariables as $variable) {
+            if ($variable['identifier'] == 'numAttempts') {
+                $attempts[(string)$this->getVariableTime($variable['epoch'])] = [];
+            }
+        }
+        foreach ($itemVariables as $variable) {
+            $cand = null;
+            $bestDist = null;
+            foreach (array_keys($attempts) as $time) {
+                $dist = abs($time - $this->getVariableTime($variable['epoch']));
+                if (is_null($bestDist) || $dist < $bestDist) {
+                    $bestDist = $dist;
+                    $cand = $time;
+                }
+            }
+            $attempts[$cand][] = $variable;
+        }
```

Comment:
```suggestion
        $attempts = [];
        
        foreach ($itemVariables as $variable) {
            if ($variable['identifier'] == 'numAttempts') {
                $attempts[(string)$this->getVariableTime($variable['epoch'])] = [];
            }
        }
        
        foreach ($itemVariables as $variable) {
            $cand = null;
            $bestDist = null;
            
            foreach (array_keys($attempts) as $time) {
                $dist = abs($time - $this->getVariableTime($variable['epoch']));
                
                if (is_null($bestDist) || $dist < $bestDist) {
                    $bestDist = $dist;
                    $cand = $time;
                }
            }
            
            $attempts[$cand][] = $variable;
        }
```

I think it is more clear when we separate lines. wdyt?

Reply:
I can't find this really helpful for readability tbh

Reply:
I see. I would recommend to give it a try. Once you start you will realise it is easy for your eyes to identify the separate contexts and understand the code. You will start feeling agony when you see piled code. 

It is like a book's page without paragraphs, you think it is good till you read a book with paragraphs. Every block express an idea, therefor should be separated.

It is just a suggestion. By experience I can say it will change in a good way :) 

I also think this method has several nested blocks, I would split them.